pixelgl/input.go

```diff
@@ -408,7 +410,24 @@ func (w *Window) UpdateInput() {
 	mainthread.Call(func() {
 		glfw.PollEvents()
 	})
+	w.doUpdateInput()
+}
+
+// WaitInput blocks until an event is received or a timeout. If timeout is 0
+// then it will wait indefinitely
+func (w *Window) WaitInput(timeout time.Duration) {
```

Comment:
I am a little concerned about the naming here - Could we potentially name this method `UpdateInputWait` so that it still conveys the intention of an update albeit potential? Totally open to other suggestions too.

Reply:
Yes, it is possible to pass a negative value here and GLFW states that timeout `must` be a positive number. I'll update the PR and default to 0 if value is less than 0.

Even though it would be worth a panic here, since the semantic of waiting for a negative might indicate a hidden programming error.

Reply:
I thought about `UpdateInputWait` but it felt wrong at first. But since someone also thinks the same, I'm okay renaming it.